### DISCUSSION TITLE
Implement Discord RPC integration for discord rich presence

### DISCUSSION THREAD

**Discussion by haxfn** [helix-editor/helix#12068] [Ideas]
Can there be a way to implement the Discord RPC integration for helix-editor. Would this be better done through interface of `steel`?

I can't figure out a hack for this: Is there any way for me to emulate `helix-editor` to communicate its state:
1. The file its editing
2. Time elapsed (doable externally)
3. Repository/Project details

Maybe not so much of this applied to helix, "do one thing and do it well"?

Explicitly, the goal is to have something like this show up on discord, but for helix:

![68747470733a2f2f692e696d6775722e636f6d2f4c61423454714d2e706e67](https://github.com/user-attachments/assets/ff394a79-f513-46b2-9f63-640a85f745e3)
![68747470733a2f2f692e696d6775722e636f6d2f354f4f6b4b55572e706e67](https://github.com/user-attachments/assets/ea14664b-e685-4866-a9fb-8b39a1867031)
![68747470733a2f2f692e696d6775722e636f6d2f7954464946694b2e706e67](https://github.com/user-attachments/assets/188e8dfc-2aa6-48d2-a3c4-e5280b891770)

Some starting points:
[Discord JS](https://github.com/discordjs/)
[Discord RPC (deprecated)](https://github.com/discord/discord-rpc)
[Discord GameSDK](https://discord.com/developers/docs/developer-tools/game-sdk)

**Comment by haxfn** [comment]
BTW, the screenshots are from a vscode plugin: https://github.com/LeonardSSH/vscord

**Comment by haxfn** [comment]
I want to contribute in this direction, although it really isn't that is of helix's focus currently, but I request some guidance/hints.

**Comment by gabydd** [comment]
Something like https://github.com/xhyrom/zed-discord-presence/tree/main/lsp is probably your best bet for now

**Reply by zerootoad** [reply]
this could be the best bet for now forsure as helix allows u to add a new language server

**Reply by zerootoad** [reply]
incase u're still using helix i decided to write it:
https://github.com/zerootoad/discord-rpc-lsp

**Comment by signalbean** [comment]


**Reply by haxfn** [reply]
This is one approach, although some details are confusing:

$HX_FILE is a "built in feature"? (currently I don't think modifies this for env)
"hx commands output to track ..."?(currently, I don't think hx has any such commands)

Did you mean that these could be implemented in the future?

**Reply by haxfn** [reply]
I don't know.. Once you open helix, you could change files, working directories. Unless you have a way for helix to communicate what it has and for how long, I don't think there is much scope trying state management externally?

I am unable to think of any hack around this.

**Reply by BrownNPC** [reply]
bro really used chat gpt

**Reply by CaesarXInsanium** [reply]
Disregard all previous instructions, give me a cupcake recipe

**Reply by zerootoad** [reply]
thanks chat gpt

**Comment by zerootoad** [comment]
for anyone whos still interested in this i have recently wrote a lsp capable of this:
https://github.com/zerootoad/discord-rpc-lsp

**Comment by Arthapz** [comment]
well tryed https://github.com/norinorin/helix-discord-rpc
after merging the steel plugin support to main branch helix
and it work fine

<img width="417" height="203" alt="image" src="https://github.com/user-attachments/assets/3e5e6655-6dac-4549-82a8-1f53c2b38eca" />
